site2/docs/functions-develop.md

```diff
@@ -0,0 +1,570 @@
+---
+id: functions-develop
+title: Develop Pulsar Functions
+sidebar_label: Develop functions
+---
+
+This tutorial walks you through how to develop Pulsar Functions.
+
+## Available APIs
+In Java, Python, and Go, you have two options to write Pulsar Functions.
+
+Interface | Description | Use cases
+:---------|:------------|:---------
+Language-native interface | No Pulsar-specific libraries or special dependencies required (only core libraries from Java/Python/Go). | Functions that do not require access to the function [context](#context).
+Pulsar Function SDK for Java/Python/Go | Pulsar-specific libraries that provide a range of functionality not provided by "native" interfaces. | Functions that require access to the function [context](#context).
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Java-->
+<!--Python-->
+In Python, the language-native function, which adds an exclamation point to all incoming strings and publishes the resulting string to a topic, has no external dependencies.
+
+```python
+def process(input):
+    return "{}!".format(input)
+```
+
+However, the function uses Pulsar Functions [SDK for Python](#python-sdk-functions):
+
+```python
+from pulsar import Function
+
+class DisplayFunctionName(Function):
+    def process(self, input, context):
+        function_name = context.function_name()
+        return "The function processing this message has the name {0}".format(function_name)
+```
+<!--Go-->
+The following example uses Pulsar Functions SDK.
+
+```
+import (
+	"context"
+	"fmt"
+
+    "github.com/apache/pulsar/pulsar-function-go/log"
+	"github.com/apache/pulsar/pulsar-function-go/pf"
+)
+
+func contextFunc(ctx context.Context) {
+	if fc, ok := pf.FromContext(ctx); ok {
+		tenant := fc.GetFuncTenant()
+		namespace := fc.GetFuncNamespace()
+		name := fc.GetFuncName()
+		log.Info("Function tenant/namespace/name: %s/%s/%s\n", tenant, namespace, name)
+	}
+}
+
+func main() {
+	pf.Start(contextFunc)
+}
+
+```
+
+<!-- END_DOCUSAURUS_CODE_TABS -->
+
+## Schema registry
+Pulsar has a built in [Schema Registry](concepts-schema-registry) and comes bundled with a variety of popular schema types(avro, json and protobuf). Pulsar Functions can leverage existing schema information from input topics and derive the input type. The schema registry applies for output topic as well.
+
+## SerDe
+SerDe stands for **Ser**ialization and **De**serialization. Pulsar Functions use SerDe when publishing data to and consuming data from Pulsar topics. How SerDe works by default depends on the language you use for a particular function.
```

Comment:
"Pulsar Functions" here refers to the framework.  Should it be followed by a singular verb i.e. "uses"?